src/main/kotlin/org/rust/lang/core/psi/RsPsiFactory.kt

```diff
@@ -4,15 +4,23 @@ import com.intellij.openapi.project.Project
 import com.intellij.psi.PsiElement
 import com.intellij.psi.PsiFileFactory
 import com.intellij.psi.PsiParserFacade
+import com.intellij.psi.tree.IElementType
 import org.rust.lang.RsFileType
+import org.rust.lang.core.psi.RsElementTypes.IDENTIFIER
+import org.rust.lang.core.psi.RsElementTypes.QUOTE_IDENTIFIER
 import org.rust.lang.core.psi.impl.mixin.selfParameter
 import org.rust.lang.core.psi.impl.mixin.valueParameters
 import org.rust.lang.core.psi.util.childOfType
 
 class RsPsiFactory(private val project: Project) {
-    fun createIdentifier(text: String): PsiElement =
-        createFromText<RsModDeclItem>("mod $text;")?.identifier
-            ?: error("Failed to create identifier: `$text`")
+    fun createIdentifier(text: String, tokenType: IElementType): PsiElement =
```

Comment:
I just tried to unify identified and quote_identified elements by providing a token type to the `RsNamedElementImpl` constructor. But that didn't work well when it came to renaming support, so I agree, maybe it's better to separate them at all levels.